comp/api/api/apiimpl/grpc.go

```diff
@@ -103,14 +105,33 @@ func (s *serverSecure) DogstatsdSetTaggerState(_ context.Context, req *pb.Tagger
 	}
 
 	// FiXME: we should perhaps lock the capture processing while doing this...
-	t := replay.NewTagger()
-	if t == nil {
+	taggerReplay := s.taggerComp.ReplayTagger()
+	if taggerReplay == nil {
 		return &pb.TaggerStateResponse{Loaded: false}, fmt.Errorf("unable to instantiate state")
 	}
-	t.LoadState(req.State)
+	state := make([]taggerTypes.Entity, len(req.State))
+
+	// better stores these as the native type
+	for id, entity := range req.State {
+		entityID, err := taggerProto.Pb2TaggerEntityID(entity.Id)
+		if err != nil {
+			log.Errorf("Error getting identity ID for %v: %v", id, err)
+			continue
+		}
+
+		state = append(state, taggerTypes.Entity{
+			ID:                          entityID,
+			HighCardinalityTags:         entity.HighCardinalityTags,
+			OrchestratorCardinalityTags: entity.OrchestratorCardinalityTags,
+			LowCardinalityTags:          entity.LowCardinalityTags,
+			StandardTags:                entity.StandardTags,
+		})
+	}
```

Comment:
I decided to do the convertion in the grpc server instead of inside the `LoadState` function, that way the replay tagger interface is free of proto dependencies